modules/core/src/system.cpp

```diff
@@ -1359,24 +1362,41 @@ void  TlsAbstraction::SetData(void *pData)
     tlsData = pData;
 }
 #else //WINRT
+static void WINAPI opencv_fls_destructor(void* pData);
```

Comment:
> WINAPI.

Should be `NTAPI`. See definition of `PFLS_CALLBACK_FUNCTION` from Windows headers.
(basically their values are equal)

Reply:
Well, in my current configuration I use **static linking**. In this case there is no DllMain at all and TLS data never cleaned up (w/o using FLS).
DllMain( DLL_THREAD_DETACH/DLL_PROCESS_DETACH, ... ) is called and every TLS (FLS) data is released as expected in **dynamic linking**.
Important note: if DLL is unloaded while thread with allocated FLS is still running, then DllMain( DLL_PROCESS_DETACH, ... ) will be called **before** any call of FLS cleanup. In this case releaseThread() will free every FLS data and no access violation occurs.